source/extensions/http/stateful_session/header/header.cc

```diff
@@ -0,0 +1,31 @@
+#include "source/extensions/http/stateful_session/header/header.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace StatefulSession {
+namespace Header {
+
+void HeaderBasedSessionStateFactory::SessionStateImpl::onUpdate(
+    const Upstream::HostDescription& host, Envoy::Http::ResponseHeaderMap& headers) {
+  absl::string_view host_address = host.address()->asStringView();
+  if (!upstream_address_.has_value() || host_address != upstream_address_.value()) {
+    const std::string encoded_address =
+        Envoy::Base64::encode(host_address.data(), host_address.length());
+    headers.addCopy(Envoy::Http::LowerCaseString(factory_.getHeaderName()), encoded_address);
```

Comment:
May be `SetCopy` here.

Reply:
thank you! That makes much more sense. Changed.